R/lru.R

```diff
@@ -30,21 +20,26 @@ LRUcache_class <- R6::R6Class("LRUcache",
       stopifnot(is.character(name) && length(name) == 1)
       stopifnot(name %in% ls(private$data))
       private$peek(name)
+    },
+    print = function() {
+      cat(paste0("<LRUcache> of capacity ", private$max_num),
+          toString(private$format_cache())
+          , sep="\n")
+      invisible(self)
     }
   ),
   private = list(
     max_num = 0,
     data    = new.env(),
-    use_bytes = FALSE,
     save    = function(name, value) {
       # Check if this value alone exceeds the cache size
-      size <- if(private$use_bytes) as.integer(as.character(pryr::object_size(value))) else 1
+      size <- private$get_new_item_size(value)
       if (size > private$max_num){
-        warning(sprintf("'%s' is too large (%dB) to fit in the cache (%dB) and will not be cached. Consider creating a larger cache.", name, size, private$max_num), call. = FALSE)
+        warning(sprintf("'%s' is too large (%d) to fit in the cache (%d) and will not be cached. Consider creating a larger cache.", name, size, private$max_num), call. = FALSE)
```

Comment:
Ah yes, I did like that too.  I'll bring it back, even if it means an if-else to turn on the units if it is a memory size (`'character'`) cache and off if it's just a number-of-items (`'numeric'`) cache.